src/App.jsx

```diff
@@ -9,14 +10,29 @@ export const tabs = [
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
-export const App = () => (
-  <div className="section">
-    <h1 className="title">Selected tab is Tab 1</h1>
+export const App = () => {
+  const [activeId, setActiveId] = useState(tabs[0].id);
 
-    <div data-cy="TabsComponent">
-      <div className="tabs is-boxed">
-        <ul>
-          <li className="is-active" data-cy="Tab">
+  function onTabSelected(id) {
+    setActiveId(id);
+  }
+
+  const selectedTab = tabs.find(tab => tab.id === activeId);
+
+  return (
+    <div className="section">
+      {/* <h1 className="title">Selected tab is Tab 1</h1> */}
+      <h1 className="title">{`Selected tab is ${selectedTab.title}`}</h1>
+
+      <div data-cy="TabsComponent">
+        <div className="tabs is-boxed">
+          <Tabs
+            tabs={tabs}
+            activeTabId={activeId}
+            onTabSelected={onTabSelected()}
```

Comment:
```suggestion
            onTabSelected={onTabSelected}
```
looks strange, you need to pass the function instead of calling it